File: classic puzzle-easy/unit-fractions.cpp
//https://www.codingame.com/training/easy/unit-fractions
//UNIT FRACTIONS
#include <bits/stdc++.h>
using namespace std;
#define LL long long int
int main()
{
    LL n;
    cin>>n;
    for(LL i=n+1;i<=2*n;i++){
       LL L=lcm(i,n);
       LL delta=L/n-L/i;
       double y=double(L)/delta;
       if(long(y)==y){
            cout<<fixed<<"1/"<<n<<" = 1/"<<long(y)<<" + 1/"<<long(i)<<endl;
       }
    }
}

File: classic puzzle-easy/unit-fractions.py
#https://www.codingame.com/training/easy/unit-fractions
#UNIT FRACTIONS
import sys
import math

n = int(input())
for x in range(n+1,2*n+1):
    l=math.lcm(n,x)
    delta=l//n-l//x
    y=l/delta
    if int(y)==y:
        s=f"1/{n} = 1/{int(y)} + 1/{x}"
        print(s)
